modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/RelatedObjectEntryResourceTest.java

```diff
@@ -0,0 +1,272 @@
+
+		ObjectRelationshipLocalServiceUtil.
+			addObjectRelationshipMappingTableValues(
+				TestPropsValues.getUserId(),
+				_objectRelationship.getObjectRelationshipId(),
+				_objectEntry.getPrimaryKey(), _user.getUserId(),
+				ServiceContextTestUtil.getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ObjectRelationshipLocalServiceUtil.
+			deleteObjectRelationshipMappingTableValues(
+				_objectRelationship.getObjectRelationshipId(),
+				_objectEntry.getPrimaryKey(), _user.getUserId());
+		ObjectRelationshipLocalServiceUtil.deleteObjectRelationship(
+			_objectRelationship);
+	}
+
+	@Test
+	public void testGetSystemObjectRelatedObjects() throws Exception {
+		JSONObject jsonObject = _invoke(
+			StringBundler.concat(
+				_userSystemObjectDefinitionMetadata.getRESTContextPath(),
+				StringPool.SLASH, _user.getUserId(), StringPool.SLASH,
+				_objectRelationship.getName()),
+			Http.Method.GET);
+
+		JSONArray itemsJSONArray = jsonObject.getJSONArray("items");
+
+		Assert.assertEquals(1, itemsJSONArray.length());
+
+		JSONObject itemJSONObject = itemsJSONArray.getJSONObject(0);
+
+		Assert.assertEquals(
+			_OBJECT_FIELD_VALUE, itemJSONObject.getString(_OBJECT_FIELD_NAME));
+	}
+
+	@Test
+	public void testPutSystemObjectRelatedObjects() throws Exception {
+		String objectFieldValue = RandomTestUtil.randomString();
+
+		ObjectEntry objectEntry = _createObjectEntry(objectFieldValue);
+
+		JSONObject jsonObject = _invoke(
+			StringBundler.concat(
+				_userSystemObjectDefinitionMetadata.getRESTContextPath(),
+				StringPool.SLASH, _user.getUserId(), StringPool.SLASH,
+				_objectRelationship.getName(), StringPool.SLASH,
+				objectEntry.getPrimaryKey()),
+			Http.Method.PUT);
+
+		Assert.assertEquals(
+			objectFieldValue, jsonObject.getString(_OBJECT_FIELD_NAME));
+	}
+
+	private ObjectEntry _createObjectEntry(String objectFieldValue)
+		throws Exception {
+
+		return ObjectEntryLocalServiceUtil.addObjectEntry(
+			TestPropsValues.getUserId(), 0,
+			_objectDefinition.getObjectDefinitionId(),
+			HashMapBuilder.<String, Serializable>put(
+				_OBJECT_FIELD_NAME, objectFieldValue
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	private ObjectDefinition _getSystemObjectDefinition(
+		SystemObjectDefinitionMetadata systemObjectDefinitionMetadata) {
+
+		List<ObjectDefinition> systemObjectDefinitions =
+			_objectDefinitionLocalService.getSystemObjectDefinitions();
+
+		for (ObjectDefinition systemObjectDefinition :
+				systemObjectDefinitions) {
+
+			if (Objects.equals(
+					systemObjectDefinition.getName(),
+					systemObjectDefinitionMetadata.getName())) {
+
+				return systemObjectDefinition;
+			}
+		}
+
+		return null;
+	}
+
+	private JSONObject _invoke(String endpoint, Http.Method httpMethod)
+		throws Exception {
+
+		Http.Options options = new Http.Options();
+
+		if (httpMethod == Http.Method.PUT) {
+			options.setPut(true);
+		}
+
+		options.addHeader(
```

Comment:
Brian might ask you to sort this alphabetically (Authorization goes first).

Reply:
I copied from [here](https://github.com/liferay/liferay-portal/blob/cb24d854af7a789ab19d470fa6bbd3fc4a51f7b5/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/graphql/test/ObjectDefinitionGraphQLTest.java#L416-L434).
I will sort the HTTP methods.